iron-list.html

```diff
@@ -1362,7 +1410,141 @@
+      model.tabIndex = currentTabIndex;
+      focusable && focusable.focus();
+    },
+
+    _manageFocus: function() {
+      var item, pidx;
+      var HIDEEN_Y = '-10000px';
+      var fidx = this._focusedIndex;
+
+      if (this._isIndexRendered(fidx)) {
+        if (this._focusedItem) {
+          item = this._getNormalizedItem(fidx);
+          pidx = this._physicalIndexForKey[this._collection.getKey(item)];
+
+          this.translate3d(0, HIDEEN_Y, 0, this._physicalItems[pidx]);
+          this._physicalItems[pidx] = this._focusedItem;
+          this._focusedItem = null;
+        }
+      } else if (!this._focusedItem) {
+        item = this._getNormalizedItem(fidx);
+        pidx = this._physicalIndexForKey[this._collection.getKey(item)];
+
+        this._focusedItem = this._physicalItems[pidx];
+        this.translate3d(0, HIDEEN_Y, 0, this._focusedItem);
+
+        if (!this._sentinelItem) {
+          var stampedTemplate = this.stamp(null);
+          this._sentinelItem = stampedTemplate.root.querySelector('*');;
+          Polymer.dom(this).appendChild(stampedTemplate.root);
+        }
+        this._physicalItems[pidx] = this._sentinelItem;
+      }
+    },
+
+    _restoreFocusedItem: function() {
+      if (this._focusedItem) {
+        Polymer.dom(this).removeChild(this._focusedItem);
+        this._focusedItem = null;
+        this._sentinelItem = null;
+      }
+    },
+
+    _didFocus: function(e) {
+      var targetModel = this.modelForElement(e.target);
+      var fidx = this._focusedIndex;
+      if (!targetModel) {
+        return;
+      }
+      if (this.modelForElement(this._focusedItem) === targetModel) {
+        this.scrollToIndex(fidx);
+        this._update();
+      } else {
+        // restore tabIndex for the currently focused item
+        this._getModelFromItem(this._getNormalizedItem(fidx)).tabIndex = -1;
+        // set the tabIndex for the next focused item
+        targetModel.tabIndex = 0;
+
+        if (fidx !== targetModel.index) {
+          this._restoreFocusedItem();
+        }
+        fidx = targetModel.index;
+        this._focusedIndex = fidx;
+        // bring the item into view
+        if (fidx < this.firstVisibleIndex || fidx > this.lastVisibleIndex) {
+          this.scrollToIndex(fidx);
+        }
+      }
+    },
+
+    _didMoveUp: function() {
+      this._focusPhysicalItem(Math.max(0, this._focusedIndex - 1));
+    },
+
+    _didMoveDown: function() {
+      this._focusPhysicalItem(Math.min(this._virtualCount, this._focusedIndex + 1));
+    },
+
+    _didEnter: function(e) {
+      var srcEvent = e.detail.keyboardEvent;
+      // toggle selection
+      this._selectionHandler(srcEvent);
+      // focus the currently focused physical item
+      this._focusPhysicalItem(this._focusedIndex);
+
+      this.fire('enter-item', null, {node: srcEvent.target})
     }
+
+    /**
+     * Fires when the user presses the enter key on an item in the list.
+     *
+     * @event enter-item

```

Comment:
Just noting that it seems inconsistent that you have `iron-list-dom-updated` qualified by the element name and this event is just `enter-item`; see note above about suggested rename to `dom-change` for that event, which may argue for keeping this one as-is, not sure.  Feel free to get other opinions.

It also seems kinda ad-hoc to fire an event for enter, but I see the use case in your demo.  It feels like it should be related to selection.  Would it make more sense to fire events related to selection/deselection (such that your demo only has to listen for selection rather than `tap` and `enter`)?